SelfAndPeerEvaluations/partner_eval_quiz_final_grader.py

```diff
@@ -0,0 +1,101 @@
+import canvasapi
+from collections import defaultdict
+import datetime
+import json
+import os
+from utils import *
+
+
+class PartnerEvalQuizGrader:
+    def __init__(
+        self,
+        course: canvasapi.course.Course,
+        json_questions: dict,
+        assignment_group: canvasapi.assignment.AssignmentGroup = None,
+    ):
+        self.course = course
+        self.json_questions = json_questions
+
+        self.assignment_group = (
+            assignment_group
+            if assignment_group is not None
+            else find_partner_eval_ag(course)
+        )
+        # make sure that assignment group has assignment attribute
+        if not hasattr(self.assignment_group, "assignments"):
+            self.assignment_group = course.get_assignment_group(
+                self.assignment_group.id, include=["assignments"]
+            )
+
+        self.student_id_map = make_student_id_map(self.course)
+
+    def upload_final_grades_to_canvas(
+        self, assignment_name="Overall Evaluation Grade"
+    ) -> None:
+        files = []
+        print("We'll be basing final grade off of these files:")
+        print("These files will be processed in proper sorted order.")
+        for filename in os.listdir("./quiz_results/quiz_reports"):
```

Comment:
The directory to search for the json files shouldn't be hard coded. It should be an argument to the `__init__` method.